kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -900,6 +910,93 @@ public void testAddingS3ObjectTags() throws Exception{
     verifyTags(expectedTaggedFiles);
   }
 
+  @Test (expected = DataException.class)
+  public void testExceptionOnNullKeys() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, null,
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, sampleHeaders());
+    writeRecordWithKeysAndHeaders(faultyRecord);
+  }
+
+  @Test (expected = DataException.class)
+  public void testExceptionOnEmptyHeaders() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, "key",
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, Collections.emptyList());
+    writeRecordWithKeysAndHeaders(faultyRecord);
+  }
+
+  @Test (expected = DataException.class)
+  public void testExceptionOnNullHeaders() throws Exception {
+    SinkRecord faultyRecord = new SinkRecord(TOPIC, PARTITION, Schema.STRING_SCHEMA, "key",
+        Schema.STRING_SCHEMA, "1", 1, 0L, TimestampType.NO_TIMESTAMP_TYPE, null);
+    writeRecordWithKeysAndHeaders(faultyRecord);
+  }
+
+  @Test
+  public void testRecordKeysAnsHeadersWritten() throws Exception {
+    writeRecordWithKeysAndHeaders(null);
+  }
+
+  private void writeRecordWithKeysAndHeaders(SinkRecord faultyRecord) throws Exception {
+    setUp();
+    // Define the partitioner
+    Partitioner<?> partitioner = new DefaultPartitioner<>();
+    partitioner.configure(parsedConfig);
+
+    // setup key record provider for writing record key files.
+    RecordWriterProvider<S3SinkConnectorConfig> keyWriterProvider =
+        new AvroFormat(storage).getRecordWriterProvider();
+    ((RecordViewSetter) keyWriterProvider).setRecordView(new KeyRecordView());
+    // setup header record provider for writing record header files.
+    RecordWriterProvider<S3SinkConnectorConfig> headerWriterProvider =
+        new AvroFormat(storage).getRecordWriterProvider();
+    ((RecordViewSetter) headerWriterProvider).setRecordView(new HeaderRecordView());
+    // initialize the KVHWriterProvider with header and key writers turned on.
+    RecordWriterProvider<S3SinkConnectorConfig> kvhWriterProvider = new KeyValueHeaderRecordWriterProvider(
```

Comment:
or simply, `writerProvider` 
With IDEs these days, there's rarely the need to use acronyms. Speaking as an ex-vim coder :)